docs/program/extend/modular-resources/_index.md

```diff
@@ -1,150 +1,528 @@
+
+from .my_base import MyBase
+
+Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new, MyBase.validate_config))
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{% /expand%}}
+
+{{% alert title="Caution" color="caution" %}}
+
+You must define all functions belonging to a built-in resource type if defining a new model. Otherwise, the class won’t instantiate.
+
+- If you are using the Python SDK, raise an `NotImplementedError()` in the body of functions you do not want to implement or put `pass`.
+- If you are using the Go SDK, return `errUnimplemented` or leave the body of functions you do not want to implement empty.
 
-A configured module can make one or more modular resources available for configuration as a component and/or service instances.
 {{% /alert %}}
 
-### Add a module to your robot configuration
+### Make your module executable
 
-The Viam module system introduces a new optional top-level configuration block to robot configurations called *modules*.
-This allows you to instruct RDK to load modules as well as register and [manage](#modular-resource-management-with-the-rdk) any modular resources made available by the module that you'd like to use with your robot.
-The RDK loads modules in the order you specify in the modules list.
+To add a module to your robot, you need to have an [executable file](https://en.wikipedia.org/wiki/Executable) that runs your module when executed, can take a local socket as a command line argument, and cleanly exits when sent a termination signal.
+Your options for completing this step are flexible, as this file does not need to be in a raw binary format.
 
-#### Required attributes - module
+One option is creating and save a new shell script (<file>.sh</file>) that runs your module.
+For example:
 
-Name | Type | Default Value | Description
--------------- | ---- | ------------- | ---------------
-`name` | string | --        | Name of the module you are registering
-`executable_path` | string | --         | The filesystem path to the module executable on the robot part
+{{< tabs name="Template Shell Scripts as Module Executables" >}}
+{{% tab name="Go" %}}
 
-### Configure a component instance for a modular resource
+``` shell
+#!/bin/sh
+cd <path-to-your-module-directory>
 
-Once you have configured a module as part of your robot configuration, you can instantiate any number of instances of a modular resource made available by that module with the component or service configuration.
-All standard properties such as *attributes* and *depends_on* are supported for modular resources.
-To correctly reference a registered modular resource, you must configure the *namespace*, *type*, *name* and *model* properties.
+go build ./
+# Be sure to use `exec` so that termination signals reach the go process,
+# or handle forwarding termination signals manually
+exec ./<your-module-directory-name> $@
+```
 
-#### Required attributes - modular component
+{{% /tab %}}
```

Comment:
This is not a great example for go. Go actually builds binaries that can be directly run. So when compiling/building the module, they'll get an actual binary. No need to try to compile it here in a shell script and then exec it.

(If you got this from RDK examples, that's a tricky workaround I set up testing, not something you'd want to do during deployment, so my apologies.)

Reply:
Ok! Addressed-- lmk if terminology is still off. Not great at go but learning